R/extract.R

```diff
@@ -195,25 +197,42 @@ extract_preprocessor.workflow <- function(x, ...) {
 
 #' @rdname extract-workflow
 #' @export
-extract_postprocessor.workflow <- function(x, estimated = TRUE, ...) {
+extract_postprocessor.workflow <- function(x, ..., estimated = FALSE) {
+  if (has_postprocessor_tailor(x)) {
+    return(extract_tailor_workflow(x, estimated = estimated))
+  }
+
+  cli_abort("The workflow does not have a postprocessor.")
+}
+
+# TODO this still needs a generic in hardat
+extract_tailor_workflow <- function(x, ..., estimated = TRUE) {
+  check_dots_empty()
+
   if (!is_bool(estimated)) {
     cli_abort(
       "{.arg estimated} must be a single {.code TRUE} or {.code FALSE}."
     )
   }
+  if (!has_postprocessor_tailor(x)) {
+    cli_abort("The workflow must have a tailor postprocessor.")
+  }
 
   if (estimated) {
     res <- x$post$fit
     if (!is.null(res)) {
       return(res)
+    } else {
+      cli_abort(c(
+        "The workflow has a tailor postprocessor, but it has not been fit yet.",
+        i = "Do you need to call {.fun fit}?"
+      ))
     }
```

Comment:
```suggestion
    if (is.null(res)) {
      cli_abort(c(
        "The workflow has a tailor postprocessor, but it has not been fit yet.",
        i = "Do you need to call {.fun fit}?"
      ))
    }

    return(res)
```

Reply:
Ah yes, thank you! We don't even need the `return()` either.